static/src/javascripts-legacy/projects/commercial/modules/third-party-tags/outbrain.js

```diff
@@ -124,52 +116,21 @@ define([
         });
     }
 
-
     function checkDependencies() {
-        return Promise.all([checkEmailSignup(), checkClashingABTest()])
-            .then(function(result) {
-
-                function findEmail(value) {
-                    return value == 'nonCompliant';
-                }
-
-                return result.find(findEmail);
-            })
-            .catch(function () {
-                return 'nonCompliant';
-            });
-    }
-
-    function checkClashingABTest() {
-        if (!clashingABTestPromise) {
-            clashingABTestPromise = new Promise(function (resolve) {
-                if (clash.userIsInAClashingAbTest()) {
-                    resolve('nonCompliant');
-                }
-                else {
-                    resolve();
-                }
-            });
-        }
-
-        return clashingABTestPromise;
-    }
-
-    function checkEmailSignup() {
-        if (!emailSignupPromise) {
-            emailSignupPromise = new Promise(function (resolve) {
-                if (config.switches.emailInArticleOutbrain &&
-                    emailRunChecks.getEmailInserted()) {
-                    // There is an email sign-up
-                    // so load the merchandising component
-                    resolve('nonCompliant');
-                } else {
-                    resolve();
-                }
-            });
-        }
+        var isCheckNonCompliant = function(result) {
+                return result;
+            };
+
+        return checkMediator.waitForCheck('isOutbrainNonCompliant')
+                    .then(function(resultList) {
+                        if (resultList.some(isCheckNonCompliant)) {
```

Comment:
little tip, you can use `resultList.some(Boolean)` and spare yourself the `isCheckNonCompliant` function.